tests/templates/test_layers/test_strongly_entangling.py

```diff
@@ -116,6 +141,14 @@ def circuit(weights):
             weights = np.random.randn(2, 2, 3, 1)
             circuit(weights)
 
+        with pytest.raises(ValueError, match="Range sequence must be of length"):
+            weights = np.random.randn(2, 2, 3)
+            circuit(weights, ranges=[1])
+
+        with pytest.raises(ValueError, match="Ranges must not be zero nor"):
+            weights = np.random.randn(1, 2, 3)
+            circuit(weights, ranges=[0])
+
```

Comment:
Sorry to be a pain - would you mind moving this to a separate test function? Now it is under "test that number of dimensions are correct" which is confusing. You could just add a function `test_exception_wrong_ranges` or so...

Reply:
Otherwise, this is great!

Reply:
Totally! Sorry for overlooking it. I can do so quickly.